### ISSUE TITLE
Move Drive Entity check box to the layers or asm root's properties 

### ISSUE THREAD

**Issue by ElementalAndy** [OurMachinery/themachinery-public#1382]
Since the motion mixer is an editor tool tab (at least how I perceived it) it is really weird UX wise to have the Drive Entity check box there since it actually affects the simulation.

![image](https://user-images.githubusercontent.com/91697782/171848050-729f5a5e-dc25-4016-8d76-def90786698f.png)

**Comment by niklas-ourmachinery** [comment]
The "Drive Entity" only affects the simulation in the Animation State Machine Preview tab. It doesn't change any properties of the Animation State Machine itself, so I think this is the right place for it.

How do you see the "Drive Entity" checkbox affecting the simulation (outside of the preview tab)?

**Comment by ElementalAndy** [comment]
Yeah, I was a bit too hasty when I wrote this issue. Sorry for that. You are totally correct that the checkbox doesn't affect anything outside of the preview tab. So in other words this is also kind of the issue here. Nothing (except in code) can change the locomotion type to TM_ANIMATION_STATE_MACHINE__LOCOMOTION__EXTERNAL for the animation state machine component.

I added combo box in the animation state machine component's properties. Do you think that is an OK solution? Shall I post a PR?
